src/order/hom/complete_lattice.lean

```diff
@@ -626,3 +626,34 @@ See also `complete_lattice_hom.set_preimage`. -/
   right_inv := λ s, by simp only [← image_comp, equiv.self_comp_symm, id.def, image_id'],
   map_rel_iff' :=
     λ s t, ⟨λ h, by simpa using @monotone_image _ _ e.symm _ _ h, λ h, monotone_image h⟩ }
+
+variables [complete_lattice α] (x : α × α)
+
+/-- The map `(a, b) ↦ a ⊔ b` as a `Sup_hom`. -/
+def sup_Sup_hom : Sup_hom (α × α) α :=
+{ to_fun := λ x, x.1 ⊔ x.2,
+  map_Sup' := λ s, begin
+    refine le_antisymm (le_Sup_iff.2 $ λ a ha, sup_le (Sup_le _) $ Sup_le _) (Sup_le _),
+    { rintro _ ⟨x, hx, rfl⟩,
+      exact le_sup_left.trans (ha $ mem_image_of_mem _ hx) },
+    { rintro _ ⟨x, hx, rfl⟩,
+      exact le_sup_right.trans (ha $ mem_image_of_mem _ hx) },
+    { rintro _ ⟨x, hx, rfl⟩,
+      exact sup_le_sup (monotone_fst $ le_Sup hx) (monotone_snd $ le_Sup hx) }
+  end }
+
+/-- The map `(a, b) ↦ a ⊓ b` as an `Inf_hom`. -/
+def inf_Inf_hom : Inf_hom (α × α) α :=
+{ to_fun := λ x, x.1 ⊓ x.2,
+  map_Inf' := λ s, begin
+    refine le_antisymm (le_Inf _) (Inf_le_iff.2 $ λ a ha, le_inf (le_Inf _) $ le_Inf _),
+    { rintro _ ⟨x, hx, rfl⟩,
+      exact inf_le_inf (monotone_fst $ Inf_le hx) (monotone_snd $ Inf_le hx) },
+    { rintro _ ⟨x, hx, rfl⟩,
+      exact (ha $ mem_image_of_mem _ hx).trans inf_le_left },
+    { rintro _ ⟨x, hx, rfl⟩,
+      exact (ha $ mem_image_of_mem _ hx).trans inf_le_right }
+  end }
```

Comment:
I think we needn't wait for #18029, but should at least incorporate the golfs [here](https://github.com/leanprover-community/mathlib/pull/18023/files).